library/public/window.h

```diff
@@ -13,9 +14,7 @@ class F3D_EXPORT window
 public:
   virtual bool update() = 0;
   virtual bool render() = 0;
-  virtual bool renderToFile(const std::string& file, bool noBackground = false) = 0;
-  virtual bool renderAndCompareWithFile(const std::string& file, double threshold,
-    bool noBackground, const std::string& outputFile = "") = 0;
+  virtual image renderToImage(bool noBackground = false) = 0;
```

Comment:
add a renderToImage(image&) ?

Reply:
I think it's useless, this would be as efficient:
```cpp
image existingImage;
existingImage = std::move(window.renderToImage());
```

Reply:
In your case there is still a copy, is there not ?

Reply:
nop, it's calling `image::oparator=(image&&)` which does a swap internally.

Reply:
ok !